libraries/appbase/application.cpp

```diff
@@ -340,6 +340,13 @@ bool application::initialize_impl(int argc, char** argv, vector<abstract_plugin*
       std::cerr << "         removing these items." << std::endl;
    }
 
+   try {
+      bpo::notify(options);
+   } catch(...) {
```

Comment:
I would catch `std::exception` instead of or in addition to the ellipsis and then print the exception message to cerr.

Reply:
I wonder if it makes any sense to make appbase depend on fc so it can use fc's exception helpers. The only reason to catch here is to make a useful error message, otherwise the exception still just makes it up to the handler in main() AFAICT. So maybe it should just print the message it currently is and rethrow?

Reply:
I'm concerned when error messages are suppressed like that, not so much in this particular case, but there's another catch-everything statement at the bottom of that method.

The method already uses a mixture of returning false and throwing exceptions for error handling, so it doesn't really matter how to report the error. I'd probably do something like `std::cerr << "Failed to notify plugins of options: " << e.what() << std::endl;` because it seems to be consistent with the majority of error handling cases in this method.